diff --git a/hw/usb/meson.build b/hw/usb/meson.build
--- a/hw/usb/meson.build
+++ b/hw/usb/meson.build
@@ -26,6 +26,9 @@ system_ss.add(when: 'CONFIG_USB_DWC2', if_true: files('hcd-dwc2.c'))
 system_ss.add(when: 'CONFIG_USB_DWC3', if_true: files('hcd-dwc3.c'))
 system_ss.add(when: 'CONFIG_USB_CHIPIDEA', if_true: files('chipidea.c'))

+system_ss.add(when: 'CONFIG_STMP3770_USBPHY', if_true: files('stmp3770_usbphy.c'))
+system_ss.add(when: 'CONFIG_STMP3770_USB', if_true: files('stmp3770_usb.c'))
+
 system_ss.add(when: 'CONFIG_IMX_USBPHY', if_true: files('imx-usb-phy.c'))
 system_ss.add(when: 'CONFIG_VT82C686', if_true: files('vt82c686-uhci-pci.c'))
 system_ss.add(when: 'CONFIG_XLNX_VERSAL', if_true: files('xlnx-versal-usb2-ctrl-regs.c'))
